src/main/java/frc/robot/Constants.java

```diff
@@ -1,10 +1,50 @@
 package frc.robot;
 
 
+import edu.wpi.first.math.MatBuilder;
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.numbers.N1;
+
 public final class Constants {
+    public static final double LOOP_PERIOD = 0.02; // loop period. [s]
+    public static final double g = 9.80665; // Gravity acceleration constant. [m/s^2]
+    public static final double UPPER_TARGET_HEIGHT = 2.64; // Height of upper target. [m]
+    public static final double PEBZNER_HEIGHT = 4.8; // Height of pebzner auditorium. [m]
+    public static final int TALON_TIMEOUT = 10; // Waiting period for configurations [ms].
+
+    public static class Shooter {
+        public static final double BALL_RADIUS = 0.12; // radius of the ball. [m]
+        public static final double BALL_WEIGHT = 0.27; // weight of the ball. [kg]
+        public static final double AREA_OF_ATTACK = Math.PI * Math.pow(BALL_RADIUS, 2); // Area that air resistance hits the ball in. [m^2]
+        public static final double Cd = 0.4; // Drag coefficient of the ball.
+        public static final double AIR_DENSITY = 1.225; // Average density of air. [kg/m^3]
+
+        public static final double NOMINAL_VOLTAGE = 12.0; // Nominal voltage. [V]
+        public static final double WHEEL_RADIUS = 1; // Radius of the flywheel. [m]
+        public static final int TICKS_PER_REVOLUTION = 2048; // Ticks per revolution of the shooter motor. [tick/c]
+        public static final double TICKS_PER_METER = Math.pow(WHEEL_RADIUS, 2) * Math.PI
+                / TICKS_PER_REVOLUTION; // Ticks per meter of the shooter flywheel. [tick/m]
 
```

Comment:
IMO these should be stored in the ports file. @adamv17, your opinion?

Reply:
I think they should be stored in the Constants class, because they don't hold information about the motor or encoder inversions and so on, instead they reflect a property of the subsystem, how many ticks will the subsystem move per meter.

Reply:
That's true, but the way I see it, the ports file is actually a "hardware configuration" file, and these seem like configurations, unlike other constants which may be different i.e. from one shooter to another.
Your choice of course.